airflow/tests/test_unit.py

```diff
@@ -107,14 +116,22 @@ def test_dag_task_ongoing_duration(aggregator, task_instance):
                 {'metadatabase': {'status': 'healthy'}, 'scheduler': {'status': 'healthy'}},
             ]
             req.get.return_value = mock_resp
-            with mock.patch(
-                'datadog_checks.airflow.airflow.AirflowCheck._get_all_task_instances',
-                return_value=task_instance.get('task_instances'),
-            ):
-                check.check(None)
 
+            if should_call_method:
+                with mock.patch(
+                    'datadog_checks.airflow.airflow.AirflowCheck._get_all_task_instances',
+                    return_value=task_instance.get('task_instances'),
+                ):
+                    check.check(None)
+            else:
+                with mock.patch(
+                    'datadog_checks.airflow.airflow.AirflowCheck._get_all_task_instances'
+                ) as mock_get_all_task_instances:
+                    check.check(None)
+                    mock_get_all_task_instances.assert_not_called()
```

Comment:
```suggestion
            with mock.patch(
                'datadog_checks.airflow.airflow.AirflowCheck._get_all_task_instances'
            ) as mock_get_all_task_instances:
                check.check(None)
                assert mock_get_all_task_instances.mock_calls == should_call_method
```
This works together with the suggestion about parameters.

Basically by saying which mock calls you expect to happen you can save yourself a layer of if-clauses and nesting. That's always a readability win.

**After you apply these changes, it makes sense to rename the variable `should_call_method`**

Reply:
Have you tested this locally? I initially tried this in my first iteration, admittedly without the parameters, and I don't believe this works. The mock.patch pass when we want to collect needs to pass the mocked return value otherwise nothing get queried, collected, and eventually turned into a metric.

Reply:
Has this worked for you locally? I think this would make sense, with some additional changes made, if we were only testing whether or not a particular method is called. However, we're doing this as well as validating the presence of the metric generated from the call. I guess your suggestion is two parts: - doing the above as well as splitting out the functionality of this method into two distinct unit tests.